src/wallet/wallet.cpp

```diff
@@ -2820,6 +2818,12 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
                     }
                 }
 
+                // If subtracting fee from recipients, we now know what fee we
+                // need to subtract, we have no reason to reselect inputs
+                if (nSubtractFeeFromAmount && nFeeRet == 0) {
```

Comment:
Good catch.  The `nFeeRet == 0` check wasn't really doing anything anyway, because pick_new_inputs doesn't ever get reset to true.  But the prior check that I edited the comment on asserts that we are always on our final pass if pick_new_inputs is false.